packages/devtools_flutter/lib/src/config.dart

```diff
@@ -10,28 +13,147 @@ import 'screen.dart';
+  }
+
+  /// Generates routes, separating the path from URL query parameters.
+  Route _generateRoute(RouteSettings settings) {
+    final uri = Uri.parse(settings.name);
+    final path = uri.path;
+
+    if (_routes.containsKey(path)) {
+      return MaterialPageRoute(settings: settings, builder: _routes[path]);
     }
-    return routeToBuilder;
+    // Return a page not found.
+    return MaterialPageRoute(
+      settings: settings,
+      builder: (BuildContext context) {
+        return DevToolsScaffold.withChild(
+          child: Center(
+            child: Text(
+              'Sorry, $uri was not found.',
+              style: Theme.of(context).textTheme.display1,
+            ),
+          ),
+        );
+      },
+    );
   }
 
-  // The mapping from routes to the collection of screens to show in the app.
-  //
-  // The /connect route will be a dependency for all the other routes.
-  final Map<String, List<Screen>> _routeToTabs = const {
-    '/': [ConnectScreen()],
-    '/connected': [
-      EmptyScreen.inspector,
-      EmptyScreen.timeline,
-      EmptyScreen.performance,
-      EmptyScreen.memory,
-      EmptyScreen.logging,
-    ],
+  /// The routes that the app exposes.
+  final Map<String, WidgetBuilder> _routes = {
+    '/': (_) => Initializer(
+          builder: (_) => const DevToolsScaffold(
+            tabs: [
+              EmptyScreen.inspector,
+              EmptyScreen.timeline,
+              EmptyScreen.performance,
+              EmptyScreen.memory,
+              EmptyScreen.logging,
+            ],
+          ),
+        ),
+    '/connect': (_) => DevToolsScaffold.withChild(child: ConnectScreenBody()),
   };
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData.light(),
+      onGenerateRoute: _generateRoute,
+    );
+  }
+}
+
+/// Widget that requires business logic to be loaded before building its
+/// [builder].
+///
+/// See [_InitializerState.build] for the logic that determines whether the
+/// business logic is loaded.
+///
+/// Use this widget to wrap pages that require [service.serviceManager] to be
+/// connected. As we require additional services to be available, add them
+/// here.
+class Initializer extends StatefulWidget {
+  const Initializer({Key key, @required this.builder})
+      : assert(builder != null),
+        super(key: key);
+
+  /// The builder for the widget's children.
+  ///
+  /// Will only be built if [_InitializerState._loaded] is true.
+  final WidgetBuilder builder;
+
+  @override
+  _InitializerState createState() => _InitializerState();
+}
+
+class _InitializerState extends State<Initializer> {
+  final List<StreamSubscription> _subscriptions = [];
+
+  /// Checks if the [service.serviceManager] is connected.
+  ///
+  /// This is a method and not a getter to communicate that its value may
+  /// change between successive calls.
+  bool _loaded = service.serviceManager.hasConnection;
```

Comment:
it seems like this comment doesn't match the code? hasConnection is a getter, not a method

Reply:
Check again.  I pushed a little eagerly :)